azkaban-common/src/main/java/azkaban/executor/AbstractExecutorManagerAdapter.java

```diff
@@ -141,9 +143,9 @@ public long getAgedQueuedFlowSize() {
     long startTime = System.currentTimeMillis();
     // TODO(anish-mal) FetchQueuedExecutableFlows does a lot of processing that is redundant, since
     // all we care about is the count. Write a new class that's more performant and can be used for
-    // metrics. this.executorLoader.fetchAgedQueuedFlows internally calls FetchQueuedExecutableFlows.
+    // metrics. this.executorLoader.selectAgedQueuedFlows internally calls SelectQueuedExecutableFlows.
     try {
-      size = this.executorLoader.fetchAgedQueuedFlows(Duration.ofMinutes(minimumAgeInMinutes))
+      size = this.executorLoader.selectAgedQueuedFlows(Duration.ofMinutes(minimumAgeInMinutes))
```

Comment:
I also just wonder this is returning size only，why not count(exe_id) from DB. Since it is also just reading all exe_ids and appending them to a list, and we get list.size()

Reply:
It is a debatable question I think...
I leave such util there for future references in case if anyone wants to inspect into the AgedQueuedFlow's ExecIds. From my perspective, such exec_id list is small enough in size.
I also wonder how should we organize the code better. For each method in DAO, we need Fetch* (for fetching the row), Select* (for selecting the ids/keys) and Count* (only for counting purpose). Too many similar methods in one class will cause some maintenance overheads and cause some difficulty to write unit test: i.e. you probably need to mock all three of such combinations if you don't want to know which method is called internally. That's why I don't introduce Count* methods in this PR. All my assumption is: Select* is good enough in performance.

Reply:
if we're selecting only running flows based on the limitations we put it is always flow amount. I agree that in all the scenarios (such as we need to know size and the exact exec_ids in the same code blocks) we should do select operation. Otherwise if I only need to know size, but I still create a list of exec_ids, (image the holdem3 outage where a n hourly flow has 200+ running flows, and it means each execution GC will have to collect multiple list with 200+ items verses only collecting an integer) in this case, the GC is "in general" faster to recycle an integer over an ArrayList.
That's why we should examine if such SIZE_USAGE_ONLY operations is frequent enough, it will make an impact.